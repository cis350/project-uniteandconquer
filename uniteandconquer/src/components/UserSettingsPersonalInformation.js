/* eslint-disable no-unused-vars */
import React, { useState } from 'react';
import SidebarSettings from './SidebarSettings';
import '../assets/UserSettingsPersonalInformation.css';

const UserDB = require('../modules/UserDB');

function UserSettingsPersonalInformation() {
  /** need to switch username and fullname to first name and last name */
  const [username, setUsername] = useState('');
  const [fullName, setFullName] = useState('');
  const [firstName, setfirstName] = useState('');
  const [lastName, setlastName] = useState('');
  const [phoneNumber, setPhoneNumber] = useState(0);
  const [email, setEmail] = useState('');
  const [tags, setTags] = useState([
    'Appliances', 'Electronics', 'Books', 'Grocery and Food',
  ]);
  const myStorage = window.sessionStorage;
  const userID = myStorage.getItem('UserID');

  const updateInformation = () => {};
  const updateTags = () => { };
  const handleFirstName = () => {
    UserDB.modifyUser(userID, 3, firstName, '', (success, err) => {
      if (!success) {
        console.log(err);
      }
    });
  };
  const handleLastName = () => {
    UserDB.modifyUser(userID, 4, lastName, '', (success, err) => {
      if (!success) {
        console.log(err);
      }
    });
  };
  const handlePhone = () => {
    UserDB.modifyUser(userID, 0, phoneNumber, '', (success, err) => {
      if (!success) {
        console.log(err);
      }
    });
  };
  const handleEmail = () => {
    UserDB.modifyUser(userID, 1, email, '', (success, err) => {
      if (!success) {
        console.log(err);
      }
    });
  };
  const handleInterests = () => {
    UserDB.modifyUser(userID, 5, tags, '', (success, err) => {
      if (!success) {
        console.log(err);
      }
    });
  };

  return (
    <div className="user-settings-personal-information">
      <SidebarSettings />
      <div>
        <div className="menu-title"><h1>Personal Information</h1></div>
        <div className="post-input">
          <div className="all-fields">
            {' '}
            <div className="post-text-fields">
              <div className="post-field">
                <div className="label">Username</div>
                <input onChange={(e) => setUsername(e.target.value)} />
                <button className="confirm" type="button" onClick={setUsername}>
                  update
                </button>
              </div>
              <div className="post-field">
                <div className="label">Full Name</div>
                <input onChange={(e) => setFullName(e.target.value)} />
                <button className="confirm" type="button" onClick={setFullName}>
                  update
                </button>
              </div>
              <div className="post-field">
                <div className="label">Phone Number</div>
                <input onChange={(e) => setPhoneNumber(e.target.value)} />
                <button className="confirm" type="button" onClick={setPhoneNumber}>
                  update
                </button>
              </div>
              <div className="post-field">
                <div className="label">Email</div>
                <input onChange={(e) => setEmail(e.target.value)} />
                <button className="confirm" type="button" onClick={setEmail}>
                  update
                </button>
              </div>
            </div>
            <div className="create-post-tags">
              <div className="tags-label">Tags</div>
              <div className="post-tags">
                {tags.map((tag) => (
                  <button type="button" className="tag">{tag}</button>
                ))}
              </div>
            </div>
          </div>
          <br />
          <div className="create-post-bottom">
            <button className="update-tags" type="button" onClick={updateTags}>
              Update Tags
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default UserSettingsPersonalInformation;
