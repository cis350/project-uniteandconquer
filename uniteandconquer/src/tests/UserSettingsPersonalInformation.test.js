/**
 * @jest-environment jsdom
 */

import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import {
  render, screen, fireEvent,
} from '@testing-library/react';
import UserSettingsPersonalInformation from '../components/UserSettingsPersonalInformation';

// https://stackoverflow.com/questions/66284286/react-jest-mock-usenavigate

const mockedUsedLink = jest.fn();
const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  Link: () => mockedUsedLink,
  useNavigate: () => mockedUsedNavigate,
}));

test('user settings for personal information content (header)', () => {
  render(<UserSettingsPersonalInformation />);
  const headerElement = screen.getByText('Personal Information');
  expect(headerElement).toBeInTheDocument();
});

test('update username', () => {
  render(<UserSettingsPersonalInformation />);
  const username = screen.getByTestId('username-input');
  fireEvent.change(username, { target: { value: 'johnnyboy0123' } });
  expect(username.value).toBe('johnnyboy0123');
  fireEvent.click(screen.getByTestId('username-update'));
});

test('update name', () => {
  render(<UserSettingsPersonalInformation />);
  const name = screen.getByTestId('name-input');
  fireEvent.change(name, { target: { value: 'johnny' } });
  expect(name.value).toBe('johnny');
  fireEvent.click(screen.getByTestId('name-update'));
});

test('update email', () => {
  render(<UserSettingsPersonalInformation />);
  const email = screen.getByTestId('email-input');
  fireEvent.change(email, { target: { value: 'johnnyboy0123@gmail.com' } });
  expect(email.value).toBe('johnnyboy0123@gmail.com');
  fireEvent.click(screen.getByTestId('email-update'));
});

// test('update ', () => {
//   render(<UserSettingsPersonalInformation />);
//   const oldPassword = screen.getByTestId('old-pw-input');
//   fireEvent.change(oldPassword, { target: { value: 'asdf!234BD3' } });
//   expect(oldPassword.value).toBe('asdf!234BD3');
//   const newPassword = screen.getByTestId('new-pw-input');
//   fireEvent.change(newPassword, { target: { value: '34321CKJF!kda' } });
//   expect(newPassword.value).toBe('34321CKJF!kda');
//   const confirmNewPassword = screen.getByTestId('confirm-new-pw-input');
//   fireEvent.change(confirmNewPassword, { target: { value: '34321CKJF!kda' } });
//   expect(confirmNewPassword.value).toBe('34321CKJF!kda');
//   fireEvent.click(screen.getByText('Change Password'));
// });

test('snapshot user settings for personal information', () => {
  const component = renderer.create(<UserSettingsPersonalInformation />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
