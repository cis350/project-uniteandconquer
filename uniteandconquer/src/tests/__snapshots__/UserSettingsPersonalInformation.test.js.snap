// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot user settings for personal information 1`] = `
<div
  className="user-settings-personal-information"
>
  <div
    className="sidebarSettings"
  >
    <div
      className="greeting"
    >
      <i
        className="far fa-user-circle fa-2x"
      />
       
      Hi, Jeremy
    </div>
    <div
      className="main-page"
      data-testid="main-page-header"
    />
    <div
      className="personal-information"
      data-testid="personal-information-header"
    />
    <div
      className="password"
      data-testid="password-header"
    />
  </div>
  <div>
    <div
      className="menu-title"
    >
      <h1>
        Personal Information
      </h1>
    </div>
    <div
      className="post-input"
    >
      <div
        className="all-fields"
      >
         
        <div
          className="post-fields"
        >
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              Username
            </div>
            <input
              data-testid="username-input"
              onChange={[Function]}
            />
            <button
              className="confirm"
              data-testid="username-update"
              onClick={[Function]}
              type="button"
            >
              update
            </button>
          </div>
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              Full Name
            </div>
            <input
              data-testid="name-input"
              onChange={[Function]}
            />
            <button
              className="confirm"
              data-testid="name-update"
              onClick={[Function]}
              type="button"
            >
              update
            </button>
          </div>
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              Phone Number
            </div>
            <input
              data-testid="phone-input"
              onChange={[Function]}
            />
            <button
              className="confirm"
              data-testid="phone-update"
              onClick={[Function]}
              type="button"
            >
              update
            </button>
          </div>
          <div
            className="post-field"
          >
            <div
              className="label"
            >
              Email
            </div>
            <input
              data-testid="email-input"
              onChange={[Function]}
            />
            <button
              className="confirm"
              data-testid="email-update"
              onClick={[Function]}
              type="button"
            >
              update
            </button>
          </div>
        </div>
        <div
          className="create-post-tags"
        >
          <div
            className="tags-label"
          >
            Tags
          </div>
          <div
            className="post-tags"
          >
            <button
              className="tag"
              type="button"
            >
              Appliances
            </button>
            <button
              className="tag"
              type="button"
            >
              Electronics
            </button>
            <button
              className="tag"
              type="button"
            >
              Books
            </button>
            <button
              className="tag"
              type="button"
            >
              Grocery and Food
            </button>
          </div>
        </div>
      </div>
      <br />
      <div
        className="create-post-bottom"
      >
        <button
          className="update-tags"
          onClick={[Function]}
          type="button"
        >
          Update Tags
        </button>
      </div>
    </div>
  </div>
</div>
`;
